packages/@aws-cdk/aws-scheduler-targets-alpha/test/kinesis-stream-put-record.test.ts

```diff
@@ -0,0 +1,530 @@
+            Effect: 'Allow',
+            Resource: {
+              'Fn::GetAtt': ['MyStream5C050E93', 'Arn'],
+            },
+          },
+        ],
+      },
+      Roles: ['someRole'],
+    });
+  });
+
+  test('creates IAM policy for imported stream with imported IAM role in the same account', () => {
+    const importedStream = kinesis.Stream.fromStreamArn(stack, 'ImportedStream', 'arn:aws:kinesis:us-east-1:123456789012:stream/Foo');
+    const importedRole = Role.fromRoleArn(stack, 'ImportedRole', 'arn:aws:iam::123456789012:role/someRole');
+
+    const streamTarget = new KinesisStreamPutRecord(importedStream, {
+      partitionKey: 'key',
+      role: importedRole,
+    });
+
+    new Schedule(stack, 'MyScheduleDummy', {
+      schedule: expr,
+      target: streamTarget,
+    });
+
+    Template.fromStack(stack).hasResource('AWS::Scheduler::Schedule', {
+      Properties: {
+        Target: {
+          Arn: 'arn:aws:kinesis:us-east-1:123456789012:stream/Foo',
+          RoleArn: 'arn:aws:iam::123456789012:role/someRole',
+          RetryPolicy: {},
+          KinesisParameters: {
+            PartitionKey: 'key',
+          },
+        },
+      },
+    });
+
+    Template.fromStack(stack).hasResourceProperties('AWS::IAM::Policy', {
+      PolicyDocument: {
+        Statement: [
+          {
+            Action: ['kinesis:ListShards', 'kinesis:PutRecord', 'kinesis:PutRecords'],
+            Effect: 'Allow',
+            Resource: 'arn:aws:kinesis:us-east-1:123456789012:stream/Foo',
+          },
+        ],
+      },
+      Roles: ['someRole'],
+    });
+  });
+
+  test('throws when stream is imported from different account', () => {
+    const stack2 = new Stack(app, 'Stack2', {
+      env: {
+        region: 'us-east-1',
+        account: '234567890123',
+      },
+    });
+
+    const importedStream = kinesis.Stream.fromStreamArn(stack2, 'ImportedStream', 'arn:aws:kinesis:us-east-1:234567890123:stream/Foo');
+    const streamTarget = new KinesisStreamPutRecord(importedStream, {
+      partitionKey: 'key',
+    });
+
+    expect(() =>
+      new Schedule(stack, 'MyScheduleDummy', {
+        schedule: expr,
+        target: streamTarget,
+      })).toThrow(/Both the schedule and the stream must be in the same account/);
+  });
+
+  test('throws when stream is imported from different region', () => {
+    const stack2 = new Stack(app, 'Stack2', {
+      env: {
+        region: 'us-west-2',
+        account: '123456789012',
+      },
+    });
+
+    const importedStream = kinesis.Stream.fromStreamArn(stack2, 'ImportedStream', 'arn:aws:kinesis:us-west-2:123456789012:stream/Foo');
+    const streamTarget = new KinesisStreamPutRecord(importedStream, {
+      partitionKey: 'key',
+    });
+
+    expect(() =>
+      new Schedule(stack, 'MyScheduleDummy', {
+        schedule: expr,
+        target: streamTarget,
+      })).toThrow(/Both the schedule and the stream must be in the same region/);
+  });
```

Comment:
```suggestion
  test.each([
    ['account', 'arn:aws: kinesis:us-east-1:999999999999:stream/Foo', /Both the schedule and the stream must be in the same account./],
    ['region', 'arn:aws: kinesis:eu-central-1:123456789012:stream/Foo', /Both the schedule and the stream must be in the same region./],
  ])('throws when Kinesis Data Stream is imported from different %s', (_, arn: string, expectedError: RegExp) => {
    const importedStream = kinesis.Stream.fromStreamArn(stack, 'ImportedStream', arn);
    const streamTarget = new KinesisStreamPutRecord(importedStream, {
      partitionKey: 'key',
    });
    expect(() =>
      new Schedule(stack, 'MyScheduleDummy', {
        schedule: expr,
        target: target,
      })).toThrow(expectedError);
  });
```
`fromStreamArn` should use the same stack.
[`fromStreamAttributes`](https://github.com/aws/aws-cdk/blob/02957dfddd895ffcca1dd485da312544babf6ccb/packages/aws-cdk-lib/aws-kinesis/lib/stream.ts#L738) will require to solve the environment for the imported stream ARN:
```
  public static fromStreamAttributes(scope: Construct, id: string, attrs: StreamAttributes): IStream {
    class Import extends StreamBase {
      public readonly streamArn = attrs.streamArn;
      public readonly streamName = Stack.of(scope).splitArn(attrs.streamArn, ArnFormat.SLASH_RESOURCE_NAME).resourceName!;
      public readonly encryptionKey = attrs.encryptionKey;
    }

    return new Import(scope, id, {
      environmentFromArn: attrs.streamArn,
    });
  }
```
Also, can you please add unit tests to verify that the environment gets resolved correctly when using `fromStreamArn` and `fromStreamAttributes` after this change?